app/controllers/int_api/marketplaces_controller.rb

```diff
@@ -46,8 +46,12 @@ def create
         locale: params[:marketplace_language]},
         marketplace[:id]).data
 
+    # make the marketplace creator land to the admin panel
+    url = "#{marketplace[:url]}/admin"
```

Comment:
More than the place the problem with this is that we're building URLs with string concatenation. This means that there's now an implicit dependency between this code and the code that builds the original url; that is the base url is something you can just append /admin to and end up with a sane result. This assumption is broken if something related to these paths is changed in routes.rb or in the code that builds the url being appended to. Rails has helpers for reverse routing (i.e. building a url) to avoid these problems and those helpers should be used whenever building URLs. I suggest making the change wherever the original url is built instead.